mantidproject/mantid
# Workbench: Project Save for plots

**Description of work.**
This is an initial implementation of Project Saving for plots, it only currently handles workspaces and plots. 

A project should now have changed when a plot is added.

(Not all plot features are saved properly in this implementation but a large amount of the settings are)

**To test:**
- Launch workbench
- Load a workspace
- Open a plot and change some cosmetic details such as line colour
- File->Save Project and choose a place to save your project folder
- File->Save Project as and choose a different place to save your project folder
- Close workbench and relaunch workbench
- File->Open Project and select the first saved project it should load those workspaces back successfully and no errors should show in the logs
- Close and relaunch workbench
- File->Open project and select the second saved project and it also should load those exact same workspaces back successfully and no errors should show in the logs and the plot should be back with the same cosmetic changes that were made. 


Fixes #24295 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
